src/github.com/matrix-org/dendrite/roomserver/storage/rooms_table.go

```diff
@@ -11,7 +12,11 @@ CREATE TABLE IF NOT EXISTS rooms (
     -- Local numeric ID for the room.
     room_nid BIGINT PRIMARY KEY DEFAULT nextval('room_nid_seq'),
     -- Textual ID for the room.
-    room_id TEXT NOT NULL CONSTRAINT room_id_unique UNIQUE
+    room_id TEXT NOT NULL CONSTRAINT room_id_unique UNIQUE,
+	-- The most recent events in the room that aren't referenced by another event.
```

Comment:
The latest_events are the events that aren't referenced by another event.
So 5 is reference by 7 so is not a latest event.
But 7 isn't referenced by anything yet so is a latest event.